classes/kernel-cve-check.bbclass

```diff
@@ -101,13 +103,23 @@ python kernel_cve_check () {
     include_ignore = d.getVar("KERNEL_CVE_CHECK_INCLUDE_IGNORE")
     python_path = d.getVar("STAGING_BINDIR_NATIVE") + "/python3-native/python3"
     remote_repo_name = d.getVar("KERNEL_CVE_CHECK_LINUX_GIT_REMOTE_REPO")
+    remote_repo_autodetect = d.getVar("KERNEL_CVE_CHECK_LINUX_GIT_REMOTE_AUTODETECT")
 
     if linux_cip_ver is None:
         bb.error("LINUX_CIP_VERSION is not set. Please set version")
         return
 
     opt_ignore = "--include-ignored" if include_ignore == "1" else ""
 
+    # Try to get remote name from kernel repository
+    if not remote_repo_name and remote_repo_autodetect == "1":
+        try:
+            bb.debug(2, "No remote name is defined and KERNEL_CVE_CHECK_LINUX_GIT_REMOTE_AUTODETECT is set. bitbake detects remote name from kernel-src")
+            stdout, _ = bb.process.run('git remote', cwd=kernel_path, shell=True)
+            remote_repo_name = stdout.splitlines()[0]
```

Comment:
I'm not sure if user has more than one remote repository but this case it doesn't work.
So, how about get remote repository name based on current branch?

When .git/config looks following contents

```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[remote "cip"]
        url = git://git.kernel.org/pub/scm/linux/kernel/git/cip/linux-cip.git
        fetch = +refs/heads/*:refs/remotes/cip/*
[remote "torvalds"]
        url = git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
        fetch = +refs/heads/*:refs/remotes/torvalds/*
[remote "stable"]
        url = git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable-rc.git
        fetch = +refs/heads/*:refs/remotes/stable/*
[branch "master"]
        remote = torvalds
        merge = refs/heads/master
[branch "test"]
        remote = cip
        merge = refs/heads/linux-4.19.y-cip
```

following commnds will show remote repository name.

```
$ current_branch=$(git branch | grep ^\* | cut -f2 -d' ')
$ git config --get branch.${current_branch}.remote
cip
```

Reply:
Thank you for the suggestion.

Your idea looks a good one in general, but in this case,
I think my way is better for the following two reasens.

1. Since users may checkout to hash or tags other than the branch

For example, when user checkouts to the tag, *current_branch* will be as follows.
Then kernel-cve-check will not be able to get remote.

```
$: git checkout v4.19.169-cip42
$: git branch | grep ^\* | cut -f2 -d' '
(HEAD
```

2. (As I understand) report_affected.py does not require remote name   

report_affected.py confirm if the remote name really exists in the repository [at here](https://gitlab.com/cip-project/cip-kernel/cip-kernel-sec/-/blob/master/scripts/kernel_sec/branch.py#L339).     
                                                                                                                 
```                                                                                                              
    for key in remotes.keys():                                                                                   
        remote = remotes[key]  # __getitem__ will add git_name                                                   
        if remote['git_name'] not in current_remotes:                                                            
            msg = "remote %r not in git repository" % remote['git_name']                                                                                                                                           
            raise argparse.ArgumentError(None, msg)                                                                                                                                                                
```                                                                                                                                                                                                                
                                                                                                                                                                                                                   
However, from what I've seen, remote['git_name'] only appears at this time.                                                                                                                                        
I'm thinking that report_affected.py may not need remote name.                                                                                                                                                     
                                                                                                                                                                                                                   
In fact, I have confirmed that cve_check passes even after the                                                                                                                                                     
following operations.
                                                                                                                                                                                                                   
```                                                                                                                                                                                                                
$: cd tmp-glibc/work-shared/qemuarm64/kernel-source                                                                                                                                                                
$: git remote set-url myorigin https://example.com                                                                                                                                                                 
$: git remote -v                                                                                                                                                                                                   
myorigin        https:/example.com (fetch)                                                                                                                                                                         
myorigin        https:/example.com (push)                                                                                                                                                                          
$: cd - && bitbake -fc cve_check linux-base                                                                                                                                                                        
```                                                                                                                                                                                                                
 
So I think the remote name passed to report_affected.py can be 
any remote that actually exists in the repository.

Reply:
> So I think the remote name passed to report_affected.py can be
> any remote that actually exists in the repository.

Okay. I'll merge it.
Thanks.

Reply:
Thanks for accepting my suggestion, too!